examples/stm32_f4ve/gui/touchscreen_calibrator.hpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2009, Martin Rosekeit
+ * Copyright (c) 2009-2010, 2012-2013, Fabian Greif
+ * Copyright (c) 2012, Niklas Hauser
+ * Copyright (c) 2012, Sascha Schade
+ * Copyright (c) 2014, Daniel Krebs
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_TOUCHSCREEN_CALIBRATOR_HPP
+#define MODM_TOUCHSCREEN_CALIBRATOR_HPP
+
+namespace modm
+{
+	class TouchscreenCalibrator
+	{
+	public:
+		TouchscreenCalibrator();
+
+		/**
+		 *
+		 */
+		bool
+		calibrate(modm::glcd::Point * display, modm::glcd::Point * sample);
+
+		void
+		translate(modm::glcd::Point * raw, modm::glcd::Point * translated);
+
+	private:
+		float An;
+		float Bn;
+		float Cn;
+		float Dn;
+		float En;
+		float Fn;
+		float scale;
+	};
+}
+
+#endif // MODM_TOUCHSCREEN_CALIBRATOR_HPP
+
+#ifndef MODM_TOUCHSCREEN_CALIBRATOR_HPP
+#	error "Don't include this file directly, use touchscreen_calibrator.hpp instead!"
+#endif
+
+modm::TouchscreenCalibrator::TouchscreenCalibrator() :
+	scale(0.f)
+{
+}
+
+bool
+modm::TouchscreenCalibrator::calibrate(
+		modm::glcd::Point * display, modm::glcd::Point * sample)
+{
+	// K��(X0��X2) (Y1��Y2)��(X1��X2) (Y0��Y2)
```

Comment:
WTF-8?
![The self-help group "WTF-8 Problems" meets this time in the g��en room!](https://user-images.githubusercontent.com/2820734/93669734-7ef14180-fa96-11ea-9e21-8257bc4f9a46.png)